src/cluster.py

```diff
@@ -95,6 +122,62 @@ def _get_postgresql_version(self) -> str:
         # Remove the Ubuntu revision from the version.
         return str(package.version).split("+")[0]
 
+    def get_primary(self, unit_name_pattern=False) -> str:
+        """Get primary instance.
+
+        Args:
+            unit_name_pattern: whether to convert pod name to unit name
+
+        Returns:
+            primary pod or unit name.
+        """
+        primary = None
+        # Request info from cluster endpoint (which returns all members of the cluster).
+        r = requests.get(f"http://{self.unit_ip}:8008/cluster")
+        for member in r.json()["members"]:
+            if member["role"] == "leader":
+                primary = member["name"]
+                if unit_name_pattern:
+                    # Change the last dash to / in order to match unit name pattern.
+                    primary = "/".join(primary.rsplit("-", 1))
+                break
+        return primary
+
+    def are_all_members_ready(self) -> bool:
+        """Check if all members are correctly running Patroni and PostgreSQL.
+
+        Returns:
+            True if all members are ready False otherwise. Retries over a period of 10 seconds
+            3 times to allow server time to start up.
+        """
+        # Request info from cluster endpoint
+        # (which returns all members of the cluster and their states).
+        try:
+            for attempt in Retrying(stop=stop_after_delay(10), wait=wait_fixed(3)):
+                with attempt:
+                    r = requests.get(f"http://{self.unit_ip}:8008/cluster")
```

Comment:
```suggestion
               
                    cluster_status = requests.get(f"http://{self.unit_ip}:8008/cluster")
```

Reply:
Variable renamed in https://github.com/canonical/postgresql-operator/pull/10/commits/0afd3c85851e1dc5c9e3350dd29cbc130b28c056.